src/perform_step/bdf_perform_step.jl

```diff
@@ -822,18 +822,18 @@ function perform_step!(integrator,cache::QNDFCache,repeat_step=false)
 
 
   if integrator.opts.adaptive
-    if cnt == 1
+    if cnt > k
```

Comment:
why >k?

Reply:
Yeah, I guess there's some rectification required. (Was done because 1st step is calculated using BDF (cnt = 1), and then NDF is used. That's why set it EEst to 1 when cnt>k).

Reply:
but after k steps it should be safe to be changing around the stepsize?

Reply:
Are you talking about `integrator.EEst`?  I assume it must be safe. But let me look into it further to make adaptivity more logical. Can you suggest something?

Reply:
this basically means stepsize won't change after 5 iterations, that's basically non-adaptive stepsizing

Reply:
> The ode15s is not allowed to change either the order or the step size until a minimum of
(k + 2) consecutive steps are given using the same order formula and the same step size. If one
step is unsuccessful, the order of the method or the step size are reduced. When the compulsory
(k + 2) successful steps are given, it is possible to change the order and the step size. In this
case, the step sizes which correspond to orders (k − 1) (for k > 1), k and (k + 1) (whenever the
maximum order defined has not been reached) are calculated.

This is from the paper, this doesn't look consistent with your implementation